doc/api/readline.md

```diff
@@ -525,6 +525,481 @@ added: v0.7.7
+    **Default:** `30`.
+  * `prompt` {string} The prompt string to use. **Default:** `'> '`.
+  * `crlfDelay` {number} If the delay between `\r` and `\n` exceeds
+    `crlfDelay` milliseconds, both `\r` and `\n` will be treated as separate
+    end-of-line input. `crlfDelay` will be coerced to a number no less than
+    `100`. It can be set to `Infinity`, in which case `\r` followed by `\n`
+    will always be considered a single newline (which may be reasonable for
+    [reading files][] with `\r\n` line delimiter). **Default:** `100`.
+  * `removeHistoryDuplicates` {boolean} If `true`, when a new input line added
+    to the history list duplicates an older one, this removes the older line
+    from the list. **Default:** `false`.
+  * `escapeCodeTimeout` {number} The duration `readline` will wait for a
+    character (when reading an ambiguous key sequence in milliseconds one that
+    can both form a complete key sequence using the input read so far and can
+    take additional input to complete a longer key sequence).
+    **Default:** `500`.
+
+The `readline.promises.createInterface()` method creates a new
+`readline.promises.Interface` instance.
+
+```js
+const readline = require('readline').promises;
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+```
+
+Once the `readline.promises.Interface` instance is created, the most common case
+is to listen for the `'line'` event:
+
+```js
+rl.on('line', (line) => {
+  console.log(`Received: ${line}`);
+});
+```
+
+If `terminal` is `true` for this instance then the `output` stream will get
+the best compatibility if it defines an `output.columns` property and emits
+a `'resize'` event on the `output` if or when the columns ever change
+([`process.stdout`][] does this automatically when it is a TTY).
+
+#### Use of the `completer` Function
+
+The `completer` function takes the current line entered by the user
+as an argument, and returns a `Promise` that resolves an `Array` with two
+entries:
+
+* An `Array` with matching entries for the completion.
+* The substring that was used for the matching.
+
+For instance: `[[substr1, substr2, ...], originalsubstring]`.
+
+```js
+function completer(line) {
+  return new Promise((resolve, reject) => {
+    const completions = '.help .error .exit .quit .q'.split(' ');
+    const hits = completions.filter((c) => c.startsWith(line));
+    // Show all completions if none found.
+    resolve([hits.length ? hits : completions, line]);
+  });
+}
+```
+
+### readline.promises.cursorTo(stream, x, y)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `stream` {stream.Writable}
+* `x` {number}
+* `y` {number}
+
+The `readline.promises.cursorTo()` method moves cursor to the specified position
+in a given [TTY][] `stream`.
+
+## readline.promises.emitKeypressEvents(stream[, interface])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `stream` {stream.Readable}
+* `interface` {readline.Interface}
```

Comment:
Maybe `{readline.promises.Interface}` with adding this type after https://github.com/nodejs/node/blob/fad3b643500dda23dbd630ef79dc8e9bd263599d/tools/doc/type-parser.js#L117